dbms/src/Server/MetricsTransmitter.cpp

```diff
@@ -31,13 +31,19 @@ MetricsTransmitter::~MetricsTransmitter()
 
 void MetricsTransmitter::run()
 {
-	setThreadName("MetricsTransmit");
-
-	/// Next minute at 00 seconds. To avoid time drift and transmit values exactly each minute.
-	const auto get_next_minute = []
-	{
-		return std::chrono::time_point_cast<std::chrono::minutes, std::chrono::system_clock>(
-			std::chrono::system_clock::now() + std::chrono::minutes(1));
+	auto & config = Poco::Util::Application::instance().config();
+	auto interval = config.getInt(config_name + ".interval", 60);
+
+	setThreadName(("MetricsTransmit " + std::to_string(interval) + "s").c_str());
+
+	const auto get_next_time = [](size_t seconds) {
+		/// Next minute at 00 seconds. To avoid time drift and transmit values exactly each minute.
+		if (!(seconds % 60))
+			return std::chrono::time_point_cast<std::chrono::seconds, std::chrono::system_clock>(
+				std::chrono::time_point_cast<std::chrono::minutes, std::chrono::system_clock>(
+					std::chrono::system_clock::now() + std::chrono::minutes(seconds / 60)));
+		return std::chrono::time_point_cast<std::chrono::seconds, std::chrono::system_clock>(
+			std::chrono::system_clock::now() + std::chrono::seconds(seconds));
```

Comment:
I think, you should also align any intervals, which length is delimiter of 60.
For example, 15-second intervals should be aligned to 15-second boundaries, to avoid time drift.